lib/charms/layer/aws.py

```diff
@@ -527,19 +547,36 @@ def _get_account_id():
 
 def _get_policy_arn(policy_name):
     """
-    Translate a short policy name into an ARN and ensure that it is loaded.
+    Translate a short or full policy name into an ARN.
     """
-    policy_name = 'charm.aws.{}'.format(policy_name)
+    if not policy_name.startswith('charm.aws.'):
+        policy_name = 'charm.aws.{}'.format(policy_name)
     account_id = _get_account_id()
     _ensure_policy(policy_name)
     return 'arn:aws:iam::{}:policy/{}'.format(account_id, policy_name)
 
 
+def _get_policy_name(policy_arn):
+    """
+    Translate a policy ARN into a short name.
+    """
+    return policy_arn.split(':')[-1].split('/')[-1][len(ENTITY_PREFIX):]
```

Comment:
This doesn't handle app specific policies, like S3.  It also doesn't strip the '.' separator.